cmd/client-gen/main.go

```diff
@@ -43,13 +44,13 @@ func init() {
 func ClientGen(cmd *cobra.Command, args []string) error {
 	ctx := context.Background()
 
-	engineConf := &engine.Config{
+	engineConf := engine.Config{
 		Workdir:      workdir,
 		ConfigPath:   configPath,
 		RunnerHost:   internalengine.RunnerHost(),
 		NoExtensions: true,
 	}
-	return engine.Start(ctx, engineConf, func(ctx context.Context, r *router.Router) error {
+	return engine.Start(ctx, engineConf, func(ctx context.Context, rec *progrock.Recorder, r *router.Router) error {
```

Comment:
Note to reviewer: you'll be seeing this a lot.

Progrock provides `FromContext` and `ToContext` but I'm not sure if we want to commit to that. I started with it, but then there was one place the recorder was dropped because we intentionally create a `context.Background()`, and that spooked me into making it explicit in a later commit. But it is _everywhere_ and now I'm seeing the value in `context.Context` again. :sweat_smile:

Let me know what you prefer! I'm really on the fence.

Reply:
+1 for `context.Context`